### ISSUE TITLE
Buggrabber/Bug Sack error reported 

### ISSUE THREAD

**Issue by Vazziera** [WowRarity/Rarity#522]
1x integer overflow attempting to store 6484948244
[string "=[C]"]: in function `format'
[string "@Rarity/Core/GUI/FauxAchievementPopup.lua"]:135: in function `ShowFoundAlert'
[string "@Rarity/Core/EventHandlers.lua"]:718: in function `OnItemFound'
[string "@Rarity/Core/Collections.lua"]:380: in function `ScanArchProjects'
[string "@Rarity/Core/Collections.lua"]:308: in function `?'
[string "@Ace3/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:114: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:114>
[string "=[C]"]: ?
[string "@Ace3/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:24: in function <...Ons/Ace3/CallbackHandler-1.0/CallbackHandler-1.0.lua:20>
[string "@Ace3/CallbackHandler-1.0-8/CallbackHandler-1.0.lua"]:59: in function `Fire'
[string "@Ace3/AceEvent-3.0-4/AceEvent-3.0.lua"]:120: in function <Ace3/AceEvent-3.0/AceEvent-3.0.lua:119>

**Comment by Duckwhale** [comment]
I  originally suspected that the missing entry for the new dragon race (see #529 ) could cause Rarity to initialize attempts to an invalid value. And that may yet be cause for concern. However, there seems to be a separate and more serious problem with the archaeology code (other than the fact it's generally terrible). Specifically, there's this line in the Collections module:

```lua
local t = select(10, GetArtifactInfoByRace(x, y))
```

If I understand the surrounding code correctly, it is supposed to return the number of attempts (solves?). But [GetArtifactInfoByRace doesn't have a tenth return value](https://wowpedia.fandom.com/wiki/API_GetArtifactInfoByRace). This means the number of attempts will be ``nil``, and adding ``nil`` to any number can explain the strange error that caused this report in the first place. 

Case closed, right? But no... there's more:

* Assuming the "Patch Changes" section on Wowpedia is accurate, there never *was* a tenth parameter. Ever
* So why did the code return this for-ever ``nil`` value? I wondered...
* Blame shows clearly, it was changed in [this commit](https://github.com/WowRarity/Rarity/commit/de28fa513a9392ebc6053e0e03c0ba684a950439), as a "fix" for a similar issue to this one
* I found [this ancient WowAce ticket](https://www.wowace.com/projects/rarity/issues/295), which seems to be the source
* Indeed users have reported that the fix worked for them. How? Were there other (related) fixes?

Need to review the changes prior to ``r659`` in detail to understand what the context was, but I don't have the time right now.

**Comment by Allara** [comment]
While I appreciate that you love to badmouth the code any chance you get, keep in mind this all worked just fine when it was
implemented and in many cases (especially with archaeology) was definitely the only way to achieve something. It’s not my nor any other person who worked on this fault that Blizzard breaks their shit every chance they get. Also keep in mind that undocumented returns from API calls are super normal. And 2018 is not “ancient” by this addon’s standards. At all.

**Comment by Duckwhale** [comment]
You're right. I apologise for voicing my (usually strong and frequently wrong) opinions publicly :)

I don't want to derail this thread with a discussion about Blizzard's questionable practices or whether 5-year-old issues should be considered ancient in this case, so let's not. I will shoot you a clarifying message on Discord since your reply seems defensive.

Your point about undocumented returns is a good one, so I will have to check that in the context of investigating the 2018 issue. It may well explain the original change, though it's unclear why this would now have broken all of a sudden. I love a good mystery...
